src/staticanalysis/bot/static_analysis_bot/report/phabricator.py

```diff
@@ -63,52 +65,80 @@ def publish(self, issues, revision):
-                        issues=coverage_issues,
-                        bug_report_url=BUG_REPORT_URL,
-                    ),
-                )
-
-            stats.api.increment('report.phabricator.issues', len(inlines))
-            stats.api.increment('report.phabricator')
-            logger.info('Published phabricator comment')
-
+            if MODE_COMMENT in self.modes:
+                self.publish_comment(revision, issues, patches)
+            else:
+                logger.info('Skipping comment mode')
+
+            if MODE_HARBORMASTER in self.modes:
+                self.publish_harbormaster(revision, issues)
+            else:
+                logger.info('Skipping harbormaster mode')
         else:
-            # TODO: Publish a validated comment ?
             logger.info('No issues to publish on phabricator')
 
         return issues, patches
 
+    def publish_comment(self, revision, issues, patches):
+        '''
+        Publish issues through Phabricator comment
+        '''
+        # Load existing comments for this revision
+        existing_comments = self.api.list_comments(revision.phid)
+        logger.info('Found {} existing comments on review'.format(len(existing_comments)))
+
+        coverage_issues = [issue for issue in issues if issue.ANALYZER == COVERAGE]
+
+        # First publish inlines as drafts
+        inlines = list(filter(None, [
+            self.comment_inline(revision, issue, existing_comments)
+            for issue in issues
+            if issue.ANALYZER not in ANALYZERS_WITHOUT_INLINES
+        ]))
+        if not inlines and not patches and not coverage_issues:
+            logger.info('No new comments found, skipping Phabricator publication')
+            return
+        logger.info('Added inline comments', ids=[i['id'] for i in inlines])
+
+        # Then publish top comment
+        non_coverage_issues = [issue for issue in issues if issue.ANALYZER != COVERAGE]
+        if len(non_coverage_issues):
+            self.api.comment(
+                revision.id,
+                self.build_comment(
+                    issues=non_coverage_issues,
+                    patches=patches,
+                    bug_report_url=BUG_REPORT_URL,
+                ),
+            )
+
+        # Then publish top coverage comment
+        if len(coverage_issues):
+            self.api.comment(
+                revision.id,
+                self.build_coverage_comment(
+                    issues=coverage_issues,
+                    bug_report_url=BUG_REPORT_URL,
+                ),
+            )
+
+        stats.api.increment('report.phabricator.issues', len(inlines))
+        stats.api.increment('report.phabricator')
+        logger.info('Published phabricator comment')
+
+    def publish_harbormaster(self, revision, issues):
+        '''
+        Publish issues through HarborMaster
+        '''
+        revision.update_status(
+            state=BuildState.Pass,
```

Comment:
You could make this BuildState.Work, since we set BuildState.Pass in `update_status`